alfred-telemetry-platform/src/main/java/eu/xenit/alfred/telemetry/binder/LicenseMetrics.java

```diff
@@ -0,0 +1,118 @@
+package eu.xenit.alfred.telemetry.binder;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import org.alfresco.service.descriptor.Descriptor;
+import org.alfresco.service.descriptor.DescriptorService;
+import org.alfresco.service.license.LicenseDescriptor;
+import org.alfresco.service.license.LicenseService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+import javax.annotation.Nonnull;
+
+public class LicenseMetrics implements MeterBinder, ApplicationContextAware {
+
+    private static final Logger logger = LoggerFactory.getLogger(LicenseMetrics.class);
+
+    private static final String METRIC_NAME_LICENSE = "license";
+
+    private ApplicationContext ctx;
+    private DescriptorService descriptorService;
+
+    public LicenseMetrics(DescriptorService descriptorService) {
+        this.descriptorService = descriptorService;
+    }
+
+    @Override
+    public void bindTo(@Nonnull MeterRegistry registry) {
+        // do not do anything for Community
+        Descriptor serverDescriptor = descriptorService.getServerDescriptor();
+        if(!"Enterprise".equals(serverDescriptor.getEdition())) {
+            logger.info("Edition={}, license metrics are not available", serverDescriptor.getEdition());
+            return;
+        }
+
+        Gauge.builder(METRIC_NAME_LICENSE + ".valid", ctx, LicenseMetrics::getValid)
+                .description("Whether the license is still valid")
+                .register(registry);
+
+        Gauge.builder(METRIC_NAME_LICENSE + ".remainingDays", descriptorService, LicenseMetrics::getRemainingDays)
+                .description("Remaining days")
+                .register(registry);
+        Gauge.builder(METRIC_NAME_LICENSE + ".maxDocs", descriptorService, LicenseMetrics::getMaxDocs)
+                .description("Max docs")
+                .register(registry);
+        Gauge.builder(METRIC_NAME_LICENSE + ".maxUsers", descriptorService, LicenseMetrics::getMaxUsers)
```

Comment:
Does the license service also have some details about the *current* number of users and docs?

(So we can alert when those numbers are getting close together)

Reply:
No it doesn't.

Reply:
Because it shows those two values on the License admin page, I expected it to maybe be present.

But if it's not, then some other metrics component can export those metrics, which is fine too.